dialogue-core/src/main/java/com/palantir/dialogue/core/WeakReducingGauge.java

```diff
@@ -48,19 +55,20 @@ public synchronized void add(T sourceElement) {
     }
 
     @Override
-    public synchronized Integer getValue() {
-        return weakSet.stream().mapToInt(gaugeFunction).sum();
+    public synchronized Number getValue() {
+        return operator.apply(weakSet.stream().mapToLong(gaugeFunction));
     }
 
-    public static <T> WeakSummingGauge<T> getOrCreate(
-            ToIntFunction<T> toIntFunction,
-            T initialValue,
+    public static <T> WeakReducingGauge<T> getOrCreate(
             TaggedMetricRegistry taggedMetricRegistry,
-            MetricName metricName) {
+            MetricName metricName,
+            ToLongFunction<T> toLongFunc,
+            Function<LongStream, Number> reducingFunction,
```

Comment:
I think this will be a bit cleaner if we leave streams out of the API, much like the standard collectors. Standard accumulator/combiner model is nice to work with.

Reply:
I dunno it's actually really concise, e.g. `LongStream::sum` or `LongStream::average`? It also lets me stick em all in an array so that I can log something if there are multiple.

Reply:
No strong opinion since this isn't a reusable library :-)